### PR TITLE
kodiPlugins: init vfs.{sftp, libarchive}

### PR DESCRIPTION
**Pull request by minijackson**
###### Motivation for this change

In the recent v18 update, Kodi moved the support for some filesystems into binary add-ons (including SFTP which I was using). Now, adding `nixpkgs.config.kodi.enableVFSSFTP = true` in the configuration brings back SFTP sources (and similar with RAR and Libarchive) \o/

I tested the SFTP with my sources and works fine, for RAR and Libarchive, the add-ons could be enabled without errors but I did not test them past that.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date (**no documentation**)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) **No LICENSE in any of these add-ons**.

### CODE REVIEW

**Comment by worldofpeace** [comment]
> In the recent v18 update, Kodi moved the support for some filesystems into binary add-ons (including SFTP which I was using). Now, adding nixpkgs.config.kodi.enableVFSSFTP = true in the configuration brings back SFTP sources (and similar with RAR and Libarchive) \o/

I think this article would describe this change pretty well for those observing:
- https://kodi.tv/article/kodi-v18-binary-add-ons-repository

**Review by worldofpeace — DISMISSED** [review]
Can you break up the addition of these packages into separate commits?
In particular we prefer to have one commit for adding yourself to the maintainers list like:
```
maintainers: add minijackson
```

The attribute to use for the plugins would be
```
kodiPlugins.plugin-name: init at version
```

**Comment by minijackson** [comment]
Done! Thank you for explaining

File: pkgs/applications/video/kodi/plugins.nix [hunk 1]
```diff
@@ -506,4 +507,64 @@ let self = rec {
...
+    src = fetchFromGitHub {
+      owner = "xbmc";
+      repo = namespace;
+      rev = "${version}-${rel}";
+      sha256 = "1p21x1ncj233qa4zsi8nazcdbh0vqr36pabldmbwsp2zd5nzqz0q";
+    };
+
+    meta = with stdenv.lib; {
+      description = "RAR Virtual Filesystem add-on for Kodi";
+      platforms = platforms.all;
```
Comment (worldofpeace) [inline pkgs/applications/video/kodi/plugins.nix]: Found what we might guess the license is at https://github.com/xbmc/vfs.libarchive/blob/2ba11021ef58cf7fa9ca1051e5c0d558cd2a04bc/debian/copyright#L18

I checked and it's the same for all of them.

```suggestion
      license = licenses.gpl2Plus;
      platforms = platforms.all;
```

**Comment by worldofpeace** [comment]
I've done the following to test:
- Build all addons locally
- Build kodi with the following config
```
nix-build -E 'with (import ./.{ config = { kodi = { enableVFSSFTP = true; enableVFSRAR = true; enableVFSLibarchive = true;  };  };  }); kodi'
```
- enabled the addons within the application

However I didn't test if I could actually use these addons.
For me to merge this I need at least some sort of verification that they're actually usable.

**Comment by minijackson** [comment]
For the SFTP test, I'm sure it works (Kodi wouldn't play any remote media before I enabled the plugin). If you want to test it yourself, you could start a local ssh server, add a new video source -> Browse -> Add Network location -> Protocol: "Secure Shell (SSH/SFTP)" -> etc.

If you need a test video that will be recognized by Kodi, you can do: `ffmpeg -f lavfi -i testsrc -t 30 'The Room (2003).mp4'`

For the RAR and Libarchive plugin, I confess I haven't been able to see how I can get Kodi to recognize an archive, but it might be just a UI issue.

Also, after looking at some issues in the vfs.rar repo, I found that there is no more need for this plugin as rar support is provided in vfs.libarchive (https://github.com/xbmc/vfs.rar/issues/29). Shall I remove it?

**Comment by worldofpeace** [comment]
> Also, after looking at some issues in the vfs.rar repo, I found that there is no more need for this plugin as rar support is provided in vfs.libarchive ([xbmc/vfs.rar#29](https://github.com/xbmc/vfs.rar/issues/29)). Shall I remove it?

Yes, I think this is proof that this is true https://github.com/xbmc/vfs.libarchive/blob/2ba11021ef58cf7fa9ca1051e5c0d558cd2a04bc/vfs.libarchive/addon.xml.in#L11

> For the RAR and Libarchive plugin, I confess I haven't been able to see how I can get Kodi to recognize an archive, but it might be just a UI issue.

I've tested this simply by archiving a video into a `.7z` and then I was able to view it within the library :+1: 
This was with only the `vfs.libarchive` addon enabled.

**Comment by minijackson** [comment]
Removed the vfs.rar commit

**Comment by worldofpeace** [comment]
Thanks @minijackson :sparkles: 

Hope you had a good time contributing to nixpkgs.

**Pull request merged**